### PR TITLE
Check that message is a string in LoggerWriter to avoid stack overflow

### PR DESCRIPTION
**Pull request by jimboca**
@Einstein42 you good with this change?  It avoids the stack overflow issue I mentioned on slack.

### CODE REVIEW

**Comment by firstone** [comment]
I don't think it's very safe:

>>> sys.version_info[0]
2
>>> type('')
<type 'str'>

I think safer way to check for type would be

if type(xxx) == type('') 

or something like that.

**Comment by jimboca** [comment]
Yes, but that has issues with unicode in python 2
https://stackoverflow.com/questions/4843173/how-to-check-if-type-of-a-variable-is-string

**Comment by firstone** [comment]
I hate branching on version in code. Do you know what types it's breaking on? It is unicode? bytes? I still think it's better to do more explicit:

if type(xxx) == type('') or type(xxx) == type(u'') or type(xxx) == type(b''):

But if you don't feel that way, keep as-is. As long as it works.

File: polyinterface/polyinterface.py [hunk 1]
```diff
@@ -25,6 +25,12 @@
 from threading import Thread
 import warnings
 
+PY2 = sys.version_info[0] == 2
+
+if PY2:
+    string_types = basestring,
```
Comment (firstone) [inline polyinterface/polyinterface.py]: do you need comma at the end of the line here?

Comment (jimboca) [reply polyinterface/polyinterface.py]: I had noticed that on the stack overflow example, and meant to pull them off both lines.

**Comment by jimboca** [comment]
> I hate branching on version in code. Do you know what types it's breaking on? It is unicode? bytes? I still think it's better to do more explicit:
> 
> if type(xxx) == type('') or type(xxx) == type(u'') or type(xxx) == type(b''):
> 
> But if you don't feel that way, keep as-is. As long as it works.

I really don't have a big opinion either way, so will let @Einstein42 make the call...

File: polyinterface/polyinterface.py [hunk 2]
```diff
@@ -35,8 +41,11 @@ def __init__(self, level):
         self.level = level
 
     def write(self, message):
-        if not re.match(r'^\s*$', message):
-            self.level(message.strip())
+        if isinstance(message, string_types):
+            if not re.match(r'^\s*$', message):
+                self.level(message.strip())
+        else:
+            self.level('ERROR: message was not a string: {}'.format(message))
```
Comment (jimboca) [inline polyinterface/polyinterface.py:49]: I've also tested with this which works, so I don't really care which is preferred.
```
    def write(self, message):
        if type(message) == type('') or type(message) == type(u'') or type(message) == type(b''):
            if not re.match(r'^\s*$', message):
                self.level(message.strip())
        else:
            self.level('ERROR: message was not a string: {}'.format(message))
```

**Comment by jimboca** [comment]
@Einstein42 Any opinion?  I have tested both and they work with python 3.

**Comment by jimboca** [comment]
@Einstein42 @exking @firstone Also added fix for SSLError discussed in the ecobee thread can you let me know your opinion on both these changes?

**Comment by exking** [comment]
@jimboca thanks, let's see how it goes :)

**Pull request merged**